### PR TITLE
[CI] Remove template injection vulnerable bits from Agent Delivery / ASC GHA Workflows

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR removes the template injection vulnerable bits from Agent Delivery GHA Workflows.

### Motivation

TL; DR: [From this remediation security post from Gitlab](https://securitylab.github.com/resources/github-actions-untrusted-input/):
The best practice to avoid code and command injection vulnerabilities in GitHub workflows is to set the untrusted input value of the expression to an intermediate environment variable:
```
- name: print title
  env:
    TITLE: ${{ github.event.issue.title }}
  run: echo "$TITLE"
```
This way, the value of the ${{ github.event.issue.title }} expression is stored in memory and used as variable instead of influencing the generation of script. As a side note, it is a good idea to double quote shell variables to avoid [word splitting](https://github.com/koalaman/shellcheck/wiki/SC2086), but this is [one](https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow) of [many](https://mywiki.wooledge.org/BashPitfalls) general recommendations for writing shell scripts, not specific to GitHub Actions.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
LGTM !

File: .github/workflows/buildimages-update.yml [hunk 1]
```diff
@@ -111,19 +116,20 @@ jobs:
         id: check_pr
         env:
           GH_TOKEN: ${{ github.token }}
+          BASE_BRANCH: ${{ github.ref }}
+          INPUT_BRANCH: ${{ inputs.branch }}
         run: |
-          base_branch=${{ github.ref }} # Get the full ref
           base_branch=${base_branch#refs/heads/} # Remove 'refs/heads/' from the ref
```
Comment (chouquette) [inline .github/workflows/buildimages-update.yml:122]: I think this is now unused and can be removed as well

File: .github/workflows/buildimages-update.yml [hunk 2]
```diff
@@ -133,19 +139,25 @@ jobs:
           GH_TOKEN: ${{ github.token }}
           PR_TITLE: "[automated] ${{ steps.update_build_images.outputs.MESSAGE }}"
           PR_LABELS: "go-update,team/agent-shared-components"
+          CURRENT_BUILDIMAGE_TAG: ${{ steps.current_buildimage_tag.outputs.BUILDIMAGE_TAG }}
```
Comment (chouquette) [inline .github/workflows/buildimages-update.yml:142]: Out of curiosity, does this qualifies as untrusted output since it comes from our own tasks?

Comment (amenasria) [reply .github/workflows/buildimages-update.yml:142]: True this should be fine but the linter (it's coming soon in the CI) that I'm using does not make any difference, so I fixed it as well !

Comment (chouquette) [reply .github/workflows/buildimages-update.yml:141]: Fair enough :)

**Review by chouquette — APPROVED** [review]

**Pull request merged**